lib/iris/_data_manager.py

```diff
@@ -264,6 +277,24 @@ def _deepcopy(self, memo, data=None, fill_value='none',
 
         return result
 
+    def _propagate_masked_data_fill_value(self):
+        """
+        Align the data manager fill-value with the real masked array
+        fill-value.
+
+        """
+        data = self._real_array
+        if ma.isMaskedArray(data):
```

Comment:
I intentionally left this check as part of the `_propagate_masked_data_fill_value` method contract. I know it's private and we have more control, but the deal is that it requires masked data. Adding this check (which is cheap) reinforces the contract of the method, and de-couples it from any "soft" requirements that it just happens to be passed the right kinda data from the caller... otherwise an exception will be raised.

I'd rather keep this check for these reasons.